gammapy/utils/scripts.py

```diff
@@ -133,23 +140,70 @@ def _configure_root_logger(level='info', format=None):
 
 
 def read_yaml(filename, logger=None):
-    """Read config from YAML file."""
+    """
+    Read config from YAML file 
+    """
     import yaml
     if logger is not None:
         logger.info('Reading {}'.format(filename))
     with open(filename) as fh:
         config = yaml.safe_load(fh)
+
     return config
 
 
 def write_yaml(config, filename, logger=None):
-    """Write YAML config file
+    """
+    Write YAML config file
 
     This function can be used by scripts that alter the users config file.
+
+    Parameters
+    ----------
+    config : dict
+        config to write
+    filename : str, `~gammapy.exter.pathlib.Path`
+        file to write
     """
     import yaml
-    filename = filename + '.yaml'
+    filename = make_path(filename)
     if logger is not None:
         logger.info('Writing {}'.format(filename))
-    with open(filename, 'w') as outfile:
+    with open(str(filename), 'w') as outfile:
         outfile.write(yaml.dump(config, default_flow_style=False))
+
+
+def make_path(path):
+    """
+    Expand environment varibles on `~pathlib.Path` construction
+
+    Parameters
+    ----------
+    path : str, `~gammapy.extern.pathlib.Path`
+        path to expand
+    """
+    return Path(expandvars(str(path)))
+
+
+def recursive_update_dict(old, new):
+    """Recursively update a dict with new values
+
+    The built-in update function cannot be used for hierarchical dicts
+    where only a subset of keys shall be updated.
+    see: http://stackoverflow.com/questions/3232943/update-value-of-a-nested-dictionary-of-varying-depth
+
+    Parameters
+    ----------
+    old : dict
+        dict to be changed
+    new : dict
+        dict containing changes
+    """
+
+    result = old.copy()
+    for k, v in new.iteritems():
```

Comment:
`.iteritems` doesn't exist in Python 3 ... use `.items` to make it work in Py 2 / 3.

Maybe use this example to add a test to make sure it works on Python 2 / 3?
```python
>>> first = dict(a=42, b=dict(c=43, e=44))
>>> second = dict(d=99, b=dict(g=98, c=50))
>>> from gammapy.utils.scripts import recursive_update_dict
>>> recursive_update_dict(first, second)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/deil/code/gammapy/gammapy/utils/scripts.py", line 204, in recursive_update_dict
    for k, v in new.iteritems():
AttributeError: 'dict' object has no attribute 'iteritems'
```